modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/http4s/WebServerLauncher.scala

```diff
@@ -1,38 +1,111 @@
+object WebServerLauncher extends ServerApp with LogInitialization {
   // Initialize the log and exit if it fails
   configLog.run.onException(IO(sys.exit(1))).unsafePerformIO()
 
+  // Initialize logger after the configuration
   val logger = Logger.getLogger(getClass.getName)
 
-  // TODO improve configuration style
-  val devMode = !args.contains("prod")
-
-  def launch(port: Int):Option[Server] = {
-    logger.info(s"Starting web server on port $port")
-    try {
-      Some(BlazeBuilder.bindHttp(port, "0.0.0.0")
-        .withWebSockets(true)
-        .mountService(StaticRoutes.service(devMode), "/")
-        .mountService(SeqexecCommandRoutes.service, "/api/seqexec/commands")
-        .mountService(SeqexecUIApiRoutes.service, "/api")
-        .run)
-    } catch {
-      case e: Throwable =>
-        e.printStackTrace()
-        None
+  /**
+    * Configuration for the web server
+    */
+  case class WebServerConfiguration(host: String, port: Int, devMode: Boolean)
+
+  /**
+    * Configuration for the seqexec engine
+    */
+  case class SeqexecConfiguration(odbHost: String)
+
+  // Attempt to get the file or throw an exception if not possible
+  val configurationFile: File = new File(new File(baseDir, "conf"), "app.conf")
+
+  // Read the config, first attempt the file or default to the classpath file
+  val config: Task[Config] = knobs.loadImmutable(
+    Required(FileResource(configurationFile) or ClassPathResource("app.conf")) :: Nil)
+
+  // configuration specific to the web server
+  val serverConf: Task[WebServerConfiguration] =
+    config.map { cfg =>
+      val host = cfg.require[String]("web-server.host")
+      val port = cfg.require[Int]("web-server.port")
+      val devMode = cfg.require[String]("mode")
+      WebServerConfiguration(host, port, devMode.equalsIgnoreCase("dev"))
+    }
+
+  // Configuration of the ldap clients
+  val ldapConf: Task[LDAPConfig] =
+    config.map { cfg =>
+      val urls = cfg.require[List[String]]("authentication.ldapURLs")
+      LDAPConfig(urls)
+    }
+
+  // Configuration of the authentication service
+  val authConf: Task[AuthenticationConfig] =
+    for {
+      ld <- ldapConf
+      cfg <- config
+    } yield {
+      val devMode = cfg.require[String]("mode")
+      val sessionTimeout = cfg.require[Int]("authentication.sessionLifeHrs")
+      val cookieName = cfg.require[String]("authentication.cookieName")
+      val secretKey = cfg.require[String]("authentication.secretKey")
+      val useSSL = cfg.require[Boolean]("authentication.useSSL")
+      AuthenticationConfig(devMode.equalsIgnoreCase("dev"), sessionTimeout, cookieName, secretKey, useSSL, ld)
+    }
+
+  val executorConf: Task[SeqexecConfiguration] =
+    config.map { cfg =>
+      val host = cfg.require[String]("seqexec-engine.odb")
+      SeqexecConfiguration(host)
     }
+
+  /**
+    * Configures the Seqexec executor
+    */
+  def seqexecExecutor: Kleisli[Task, SeqexecConfiguration, Unit] = Kleisli { conf =>
```

Comment:
It is unrelated to `http4s`. This is my first attempt at using `Kliesli` and I'm following `knobs` recommendations:

> Where possible, we recommend designing your applications as a free monad or use a reader monad transformer like Kleisli[Task,Config,A] to "inject" your configuration to where it's needed



On the `server` function I read the configuration for each "subsystem" and run them in the main for loop